crates/tx/src/event.rs

```diff
@@ -204,6 +206,33 @@ impl From<MaspEvent> for Event {
     }
 }
 
+/// An event that indicates the wasm payload of a tx
+pub struct TxWasmEvent {
+    /// Hash of wrapper tx
+    pub hash: Option<Hash>,
+    /// Hash of inner tx
+    pub inner_tx_hash: Hash,
+    /// The name of the wasm payload
+    pub name: String,
+}
+
+impl EventToEmit for TxWasmEvent {
+    const DOMAIN: &'static str = "tx";
+}
+
+impl From<TxWasmEvent> for Event {
+    fn from(tx_wasm_event: TxWasmEvent) -> Self {
+        let composite =
+            Self::new(EventType::new("tx-wasm-name"), EventLevel::Tx)
+                .with(CodeName(tx_wasm_event.name))
+                .with(InnerTxHash(tx_wasm_event.inner_tx_hash));
+        match tx_wasm_event.hash {
+            None => composite.into(),
+            Some(hash) => composite.with(TxHash(hash)).into(),
+        }
```

Comment:
use `.unwrap_or_default()` to get a zero hash of the wrapper, if it doesn't exist